examples/contexts/README.md

```diff
@@ -0,0 +1,61 @@
+# Using Contexts to share configuration & state
+
+A Panel component can be configured with Contexts to share configuration & state in its DOM tree instead of manually passing them into each component as props/attrs.
+
+To configure a component to provide contexts to all components in its DOM tree (including itself), set `defaultContexts` in its `config` object:
+```js
+get config() {
+  return {
+    defaultContexts: {
+      counter: new Counter(),
+      darkMode: true,
+    },
+  };
+}
+```
+
+Components must declare the names of any contexts it intends to use in its `config` object:
+```js
+get config() {
+  return {
+    contexts: [`counter`, `darkMode`],
+  };
+}
+```
+
+Once you've declared the context names, you can retrieve a context object/value by calling `Component.getContext(contextName)` (or `$component.getContext(contextName)` in jade):
+```jade
+-- /* index.jade */
+div Current count is #{$component.getContext(`counter`).getCount()})
```

Comment:
Hmm this bird may have flown the coop in jsx land, but at least in Jade templates I'd recommend passing references via `$helpers` rather than referencing `$component` directly in the template. It helps contain and specify the interface of what's exposed to the template. Back in the bad old days people occasionally tried weird abusive things like making component `update()` calls as part of template rendering code, and we'd been planning to remove the `$component` reference altogether to discourage stuff like that. Hasn't been an issue in a long time but it's why that `$helpers` distinction exists as opposed to just messing with the component/this directly in the template.

Reply:
Makes sense, I cleaned up the examples to use `$helpers` instead.